validator/client/attest.go

```diff
@@ -222,12 +222,18 @@ func (v *validator) saveAttesterIndexToData(data *ethpb.AttestationData, index u
 	return nil
 }
 
-// waitToSlotOneThird waits until one third through the current slot period
-// such that head block for beacon node can get updated.
+// waitToSlotOneThird waits until (a) or (b) whichever comes first:
+//   (a) the validator has received a valid block that is the same slot as input slot
+//   (b) one-third of the slot has transpired (SECONDS_PER_SLOT / 3 seconds after the start of slot)
 func (v *validator) waitToSlotOneThird(ctx context.Context, slot uint64) {
 	ctx, span := trace.StartSpan(ctx, "validator.waitToSlotOneThird")
 	defer span.End()
 
+	// Don't need to wait if requested slot is the same as highest valid slot.
+	if slot == v.highestValidSlot {
+		return
+	}
```

Comment:
```suggestion
	// Don't need to wait if requested slot is the same as highest valid slot.
	if slot <= v.highestValidSlot {
		return
	}
```
Perhaps we check lte just in case?